README.md

```diff
@@ -60,22 +76,53 @@ val value by intSort
+With this DSL, you can
+* transform expressions from the solver native representation to KSMT representation and vice versa,
+* use _formula introspection_,
+* manipulate expressions without involving a solver,
+* use expressions even if the solver is freed.
+
+Expression interning (hash consing) affords faster expression comparison: we do not need to compare the expression
+trees. Expressions are deduplicated, so we avoid redundant memory allocation.
+
+### Utilities to simplify and transform expressions
+
+KSMT provides a simplification engine applicable to all supported expressions for all supported theories in a
+solver-agnostic model:
+
+* reduce expression size and complexity
+* evaluate expressions (including those with free variables) — reduce your expression to a constant
+* perform formula transformations
+* substitute expressions
+
+KSMT simplification engine implements more than 200 rules.
+By default, it attempts to apply simplifications when you create the expressions, but you can turn this
+feature off, if necessary. You can also simplify an arbitrary expression manually.
+
+### Using multiple solvers (portfolio mode)
+
+SMT solvers may differ in performance across different formulas:
+see the [International Satisfiability Modulo Theories Competition](https://smt-comp.github.io/2022/).
+
+With KSMT portfolio solving, you can run multiple solvers in parallel on the same formula — until you get the first
+(the fastest) result.
+
+For detailed instructions on running multiple solvers, see [Advanced usage](docs/advanced-usage.md).
+
+### Running solvers in separate processes
+
+Most of the SMT solvers are research projects — they are implemented via native libraries and are sometimes not 
+production ready:
+* they may ignore timeouts — they sometimes hang in a long-running operation, and you cannot interrupt it;
+* they may suddenly crash interrupting the entire process — because of a pointer issue, for example.
+
+KSMT runs each solver in a separate process, which adds to stability of your application and provides support for
+portfolio mode.
 
-### Process based solver runner
-KSMT provides a high-performant API for running SMT solvers in separate processes.
-This feature is important for implementing hard timeouts and 
-solving using multiple solvers in portfolio mode.
+### KSMT distribution
 
-### Formula simplification and manipulation utils
-KSMT provides a simplification engine that can simplify and especially evaluate all supported expressions in all
-supported theories.
-Also, KSMT provides utilities for performing formula transformation, visiting and expression substitution.
+Many solvers do not have prebuilt binaries, or binaries are for Linux only.
 
-Check out our [advanced features tutorial](docs/advanced-usage.md) for the examples.
+KSMT is distributed as JVM library with solver binaries provided.
+SMT-LIB benchmarks based test environment as well as documentation and examples are also available.
```

Comment:
The test environment is not actually part of the library distribution. It is more about quality assurance.